decks/service-worker.mdx

```diff
@@ -0,0 +1,147 @@
+import { Head } from 'mdx-deck'
+
+export { default as theme } from '../theme'
+
+<Head>
+  <title>Service Worker</title>
+  <meta name="author" content="Tobias Linkohr" />
+</Head>
+
+# Service Worker
+
+by Tobias Linkohr  
+([github](https://github.com/Tobi-mmt))
+
+
+---
+
+# What is it?
+programmable network proxy
+
+---
+
+![](./images/service-worker/sw.jpeg)
+
+---
+
+# what can we do ?
+
+- receive push notifications
+- use app offline
+- cache management
+
+---
+
+# Good to know
+
+- The Service Worker runs in its __own global script context__
+- It isn’t tied to a specific web page
+- It __can't__ access the DOM
+
+---
+
+# Lifecycle
+
+![](./images/service-worker/lifecycle.png)
+
+---
+
+# use a service-worker (sw) 
+# step by step
+
+---
+
+# 1. Register SW
+
+```js
+if ('serviceWorker' in navigator) {
+  window.addEventListener('load', function() {
+    navigator.serviceWorker.register('/sw.js').then(function(registration) {
+      // Registration was successful
+      console.log('ServiceWorker registration successful');
+    }, function(err) {
+      // Registration failed
+      console.log('ServiceWorker registration failed: ', err);
+    });
+  });
+}
+```
+---
+
+# 2. Activate SW
+
+Browser activates the SW after installation
+
+
+---
+
+![](./images/service-worker/waiting.png)
```

Comment:
Could you change this to <img src="https://.." /> please. This is needed for the build.